### PR TITLE
Feat/preparing for pub dev

### PR DESCRIPTION
**Pull request by kjawadDeveloper2**
1) Added home page for the pub dev.
2) Added plugin description.
3) Removed un used files and classes.
4) Formatted the code according the pub dev suggestion.

### CODE REVIEW

**Review by haybankz — APPROVED** [review]

File: lib/src/features/helper/dec_string.dart [hunk 1]
```diff
@@ -8,15 +8,15 @@ class DecString {
...
   static const ONE = 100000000000000000.0;
 
-  static String decStringFromDouble (double f) {
+  static String decStringFromDouble(double f) {
     return (f * ONE).toStringAsPrecision(PRECISION).split('.').first;
   }
 
-  static double doubleFromDecString (String f) {
+  static double doubleFromDecString(String f) {
     var i = int.parse(f);
```
Comment (mustafapylons) [inline lib/src/features/helper/dec_string.dart:16]: same here, counter, or anything that indicates what this variable will be used for is better than "i".

File: lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart [hunk 2]
```diff
@@ -33,31 +51,40 @@ class GenesisState extends $pb.GeneratedMessage {
...
   }
-  factory GenesisState.fromBuffer($core.List<$core.int> i, [$pb.ExtensionRegistry r = $pb.ExtensionRegistry.EMPTY]) => create()..mergeFromBuffer(i, r);
-  factory GenesisState.fromJson($core.String i, [$pb.ExtensionRegistry r = $pb.ExtensionRegistry.EMPTY]) => create()..mergeFromJson(i, r);
-  @$core.Deprecated(
-  'Using this can add significant overhead to your binary. '
-  'Use [GeneratedMessageGenericExtensions.deepCopy] instead. '
-  'Will be removed in next major version')
+  factory GenesisState.fromBuffer($core.List<$core.int> i,
+          [$pb.ExtensionRegistry r = $pb.ExtensionRegistry.EMPTY]) =>
+      create()..mergeFromBuffer(i, r);
```
Comment (mustafapylons) [inline lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart:56]: same here, better to use more clear variables than "i", "r"

File: lib/src/generated/cosmos/authz/v1beta1/authz.pb.dart [hunk 3]
```diff
@@ -28,43 +39,70 @@ class GenericAuthorization extends $pb.GeneratedMessage {
...
-  static GenericAuthorization getDefault() => _defaultInstance ??= $pb.GeneratedMessage.$_defaultFor<GenericAuthorization>(create);
+  static GenericAuthorization getDefault() => _defaultInstance ??=
+      $pb.GeneratedMessage.$_defaultFor<GenericAuthorization>(create);
   static GenericAuthorization? _defaultInstance;
 
   @$pb.TagNumber(1)
   $core.String get msg => $_getSZ(0);
   @$pb.TagNumber(1)
-  set msg($core.String v) { $_setString(0, v); }
+  set msg($core.String v) {
```
Comment (mustafapylons) [inline lib/src/generated/cosmos/authz/v1beta1/authz.pb.dart:73]: does "v" stand for "validator" or similar context?

**Review by mustafapylons — COMMENTED** [review]
Overall code's readable, understandable. 
Would be better as long as you replace all variables i,r,v with meaningful, descriptive names.

File: lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart (hunk 2 above)
Comment (MikeSofaer) [reply lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart:56]: IMO "i" for index is totally standard, and not a problem.

File: lib/src/generated/cosmos/authz/v1beta1/authz.pb.dart (hunk 3 above)
Comment (kjawadDeveloper2) [reply lib/src/generated/cosmos/authz/v1beta1/authz.pb.dart:73]: Sir this is autogenerated code from the proto files.

File: lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart (hunk 2 above)
Comment (kjawadDeveloper2) [reply lib/src/generated/cosmos/auth/v1beta1/genesis.pb.dart:56]: Same auto generated code.

File: lib/src/features/helper/dec_string.dart (hunk 1 above)
Comment (kjawadDeveloper2) [reply lib/src/features/helper/dec_string.dart:16]: @afti-githobo  written code.

File: lib/src/features/helper/dec_string.dart [hunk 4]
```diff
@@ -8,15 +8,15 @@ class DecString {
   static const MAX = 999999999999999999;
   static const ONE = 100000000000000000.0;
 
-  static String decStringFromDouble (double f) {
+  static String decStringFromDouble(double f) {
     return (f * ONE).toStringAsPrecision(PRECISION).split('.').first;
```
[the remark this answers is not in the document]
Comment (kjawadDeveloper2) [reply lib/src/features/helper/dec_string.dart:12]: Afti written code.

File: lib/src/features/data/models/transaction.dart [hunk 5]
```diff
@@ -141,7 +170,7 @@ enum TxState { notYetSent, notYetAccepted, committed, refused }
 extension TxStatePar on String {
   TxState toTxStateEnum() {
     return TxState.values.firstWhere(
-            (e) => e.toString().toLowerCase() == 'TxState.$this'.toLowerCase(),
+        (e) => e.toString().toLowerCase() == 'TxState.$this'.toLowerCase(),
```
[the remark this answers is not in the document]
Comment (kjawadDeveloper2) [reply lib/src/features/data/models/transaction.dart:173]: Afti written code.

**Comment by afti-githobo** [comment]
i'll review this in full, but fwiw - running dart format on the autogenerated code feels wrong? you've got 209 file changes here and anything that touches something autogenerated is just noise, right? like, we never actually make changes to the functionality of those files, and we don't _care__ if they're readable or not. just makes the meaningful parts of the pr harder to review.

**Comment by kjawadDeveloper1** [comment]
> i'll review this in full, but fwiw - running dart format on the autogenerated code feels wrong? you've got 209 file changes here and anything that touches something autogenerated is just noise, right? like, we never actually make changes to the functionality of those files, and we don't _care__ if they're readable or not. just makes the meaningful parts of the pr harder to review.

I know but the thing is that's what the pub dev wants. I tried to work around it but they are pretty serious about the format.

**Comment by afti-githobo** [comment]
> 
> 
> > i'll review this in full, but fwiw - running dart format on the autogenerated code feels wrong? you've got 209 file changes here and anything that touches something autogenerated is just noise, right? like, we never actually make changes to the functionality of those files, and we don't _care__ if they're readable or not. just makes the meaningful parts of the pr harder to review.
> 
> I know but the thing is that's what the pub dev wants. I tried to work around it but they are pretty serious about the format.

well, if pub.dev wants stupid things, ig pub.dev gets those stupid things. no objection on that front, then.

**Review by mustafapylons — APPROVED** [review]

**Pull request merged**